9.traverseTreeAlphabetical.js

```diff
@@ -18,15 +18,25 @@
 //     this.right = null;
 //   }
 // }
-// 
+//
 // f is the root, so therefore
-// traverseTreeAlphabetical(f) Should return the array 
+// traverseTreeAlphabetical(f) Should return the array
 // [ "A", "B", "C", "D", "E", "F", "G", "H", "I", "J", "K" ]
 
 // Implement your code here
 const traverseTreeAlphabetical = (node, result = []) => {
+  if (!node) return [];
+  let n = [node];
+  let currentNode = n.shift();
+  while (n.length) {
+    result.push(currentNode.val);
+    if (currentNode.left) {
+      n.push(currentNode.left);
+    } else {
+      n.push(currentNode.right);
+    }
+  }
+  return result;
+};
```

Comment:
This looks closer to a BFS than the DFS you would need. Go back and revisit in order tree traversal. Also b/c you shift before the while loop, you never enter the loop because the array is already empty by then.